src/key.rs

```diff
@@ -1850,121 +1843,176 @@ mod test {
             "xx0218845781f631c48f1c9709e23092067d06837f30aa0cd0544ac887fe91ddd1"
         )
         .is_err());
+    }
 
+    #[test]
+    fn keys_from_str_long_invalid() {
         let long_str = "a".repeat(1024 * 1024);
         assert!(SecretKey::from_str(&long_str).is_err());
         assert!(PublicKey::from_str(&long_str).is_err());
     }
 
     #[test]
-    // In fuzzing mode the Y coordinate is expected to match the X, so this
-    // test uses invalid public keys.
+    #[cfg(all(feature = "std", not(fuzzing)))]
+    fn erased_keypair_is_valid() {
+        let s = Secp256k1::new();
+        let kp = KeyPair::from_seckey_slice(&s, &[1u8; constants::SECRET_KEY_SIZE])
+            .expect("valid secret key");
+        let mut kp2 = kp;
+        kp2.non_secure_erase();
+        assert!(kp.eq_fast_unstable(&kp2));
+    }
+
+    // FIXME(Tobin): I don't know what this is testing? There is no assertion or panic?
```

Comment:
I believe this tests that if an RNG generates an out-of-range key it is called again instead of panicking.

Reply:
oooh, thanks man.